File: 5-longest-palindromic-substring/5-longest-palindromic-substring.java
class Solution {
    public int low = 0, high = 1;
    public String longestPalindrome(String s) {
        for(int index = 0; index < s.length(); index++){
            isPalindrome(s, index, index);
            isPalindrome(s, index, index+1);
        }
        return s.substring(low, low+high);
    }
    public void isPalindrome(String str, int left, int right){
        while(left >= 0 && right < str.length() && str.charAt(left) == str.charAt(right)){
            left--;
            right++;
        }
        if(high < right-left-1){
            low = left+1;
            high = right-left-1;
        }
    }
}

File: 5-longest-palindromic-substring/5-longest-palindromic-substring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    var max = '';
    for(var index = 0; index < s.length; index++){
        for(var jIndex = 0; jIndex < 2; jIndex++){
            var left = index, right = index + jIndex;
            while(s[left] && s[left] === s[right]){
                left--;
                right++;
            }
            if((right-left-1) > max.length){
                max = s.substring(left+1, right);
            }
        }
    }
    return max;
};